subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/ConfigurationInternal.java

```diff
@@ -88,8 +89,15 @@ enum InternalState {
 
     boolean isCanBeMutated();
 
+    /**
+     * Lock the configuration, with the option to be lenient on some failures
```

Comment:
This method no longer allows you to choose the lenient option. This information should probably be documented on the new method below.